packages/fluentui/docs/src/prototypes/VirtualizedStickyTree/VirtualizedStickyTree.tsx

```diff
@@ -0,0 +1,465 @@
+import * as React from 'react';
+import { VariableSizeList as List } from 'react-window';
+import { ObjectShorthandCollection, Tree, TreeItemProps, treeTitleClassName } from '@fluentui/react-northstar';
+import StickyTreeTitle from './StickyTreeTitle';
+import { InnerElementContext, InnerElementContextType } from './context';
+
+class ItemWrapper extends React.PureComponent<{
+  index: number;
+  isScrolling?: boolean;
+  style: Object;
+  data: any;
+}> {
+  render() {
+    const { index, style, data } = this.props;
+    const item = data.renderedItems[index];
+    if (item) {
+      return data.isNonSticky(item)
+        ? React.cloneElement(item, {
+            style,
+            onTitleClick: data.onClick,
+            onFocusParent: data.handleFocusParent,
+            onFocusFirstChild: data.handleFocusFirstChild,
+            onSiblingsExpand: (event: React.SyntheticEvent<HTMLElement>, treeItemProps: TreeItemProps) => {
+              if (treeItemProps.level !== 1)
+                // don't care about onsibling Expand for sticky headers
+                data.handleOnSiblingsExpand(event, treeItemProps);
+            },
+            onFocus: (e: React.SyntheticEvent<HTMLElement>) => {
+              if (item.props.level === 1) return;
+              data.handleOnFocus(e);
+            },
+          })
+        : null;
+    }
+
+    return null;
+  }
+}
+
+const InnerElementType = ({ children, style }) => {
+  const context = React.useContext(InnerElementContext);
+  const { height, stickyItemSize, stickyTopIds, stickyBottomIds, onClickSticky } = context;
+
+  if (!children.length) return <></>;
+
+  const data = children[0]?.props?.data;
+  const { renderedItems } = data;
+
+  const stickyTopItemsIndex = stickyTopIds.map((id: string) =>
+    renderedItems.findIndex((item: React.ReactElement<TreeItemProps>) => item.props.id === id),
+  );
+  const stickyBottomItemsIndex = stickyBottomIds.map((id: string) =>
+    renderedItems.findIndex((item: React.ReactElement<TreeItemProps>) => item.props.id === id),
+  );
+
+  return (
+    <div
+      style={{
+        ...style,
+        height: Math.max(height, style.height),
+      }}
+    >
+      {stickyTopItemsIndex.map((index: number) => {
+        const item = renderedItems[index];
+        return React.cloneElement(item, {
+          style: {
+            position: 'sticky',
+            height: stickyItemSize,
+            zIndex: 2,
```

Comment:
zIndexes can be quite hard to maintain around the place, should a default be provided in some kind of theme ?

Reply:
I see your point. I'm not sure how we handle the css variables for prototypes. I will look into it and get back to this comment :)

Reply:
search for `zIndexes` in packages/fluentui/react-northstar/src/themes/teams/siteVariables.ts, maybe we can use some constants from there

Reply:
Updated in this [commit](https://github.com/microsoft/fluentui/pull/15461/commits/142bcb327b17452d9cb4894a258296214fdd558b)